### DISCUSSION TITLE
Packaging hybrid rust+python software with maturin

### DISCUSSION THREAD

**Discussion by uncomfyhalomacro** [PyO3/maturin#2182] [Q&A]
Hi. New to maturin build tool. I am a packager for openSUSE.

One of our members, @firstyear, is an employee for SUSE (not to be mistaken for openSUSE even if the two are related) and has been valuable in improving the packaging of Rust software in both openSUSE and SUSE. Their proposed solution is to include the following environmental variables in the build environment

```
export CARGO_NET_OFFLINE=true
export CARGO_PROFILE_RELEASE_DEBUG=full
export CARGO_PROFILE_RELEASE_SPLIT_DEBUGINFO=off
export CARGO_PROFILE_RELEASE_STRIP=false
```

I would love to ask if maturin does the following things that I might be incorrect:
- does maturin override `CARGO_*` environmental variables e.g. with `maturin.tool.strip = true`?
- will it respect `RUSTFLAGS` even if `strip = true` is set in `pyproject.toml`?

I asked this because it's part of openSUSE's policy to have debuginfo for builds (?).

I will wait for your response! Thank you!

**Comment by messense** [comment]
> * does maturin override `CARGO_*` environmental variables e.g. with `maturin.tool.strip = true`?

For `strip` it only add `-C strip=symbols` to `RUSTFLAGS`, the only `CARGO_*` env var overridden is `CARGO_ENCODED_RUSTFLAGS` which shouldn't be set by normal users/packagers. So generally speaking it does not override `CARGO_*` env vars.

> * will it respect `RUSTFLAGS` even if `strip = true` is set in `pyproject.toml`?

I'm guessing no because it currently [appends `-C strip=symbols` to `RUSTFLAGS`](https://github.com/PyO3/maturin/commit/ed8ba1e2f3dccabd969c2500ac4330596d96641a), but it shouldn't be hard to change to respect existing strip options in `RUSTFLAGS`.

**Reply by Firstyear** [reply]
@uncomfyhalomacro So we need to make sure we don't set maturin.tool.strip = true, otherwise we should be good to set the other env vars in cargo-packaging.

**Comment by Thaodan** [comment]
Just encountered the same thing.. Not sure if it's a duplicate: #2951
